src/tests/shamrock/patch/PatchDataLayerLayoutTests.cpp

```
@@ -44,3 +45,20 @@ NEW_TEST(Unittest, "shamrock/patch/PatchDataLayerLayout::serialize_json", 1) {
+NEW_TEST(Unittest, "shamrock/patch/PatchDataLayerLayout::add_field", 1) {
+    using namespace shamrock::patch;
+
+    PatchDataLayerLayout pdl;
+
+    u32 nfields = 0;
+#define X(type)                                                                                    \
+    pdl.add_field<type>(#type, 1);                                                                 \
+    nfields++;
+    XMAC_LIST_ENABLED_FIELD
+#undef X
+
```

Cppcheck (2.21.0)

[style] The function 'assert_equal' is never used.

(unusedFunction)